Assets/Scripts/Entities/Player/WeaponHandler.cs

```diff
@@ -13,16 +13,21 @@ public class WeaponHandler : MonoBehaviour
         private GameObject _currentWeapon;
         private GameObject _selectedWeapon;
         private AnimationHandler _animHandler;
+        private bool isPlayer;
 
         public void Start()
         {
             _animHandler = GetComponent<AnimationHandler>();
+            isPlayer = GetComponent<Entity>().tag == "Player";
         }
 
         public void Update()
         {
-            if (CheckForWeapon() && Input.GetKeyDown(KeyCode.E))
-                SwitchWeapon();
+            if (isPlayer)
```

Comment:
Waarom deze check? De checkforweapon method baseert de check op de muis positie die altijd in het midden is. dus getw erkt alleen maar op de player